### PR TITLE
chore: Restore @aws-cdk/aws-serverless with redirect

### LINKED ISSUE #2092: Deprecate "@aws-cdk/aws-serverless"
Something that we missed in #2074 is to "deprecate" @aws-cdk/aws-serverless instead of deleting it, at least for one release, in order for users to migrate over to @aws-cdk/aws-sam and not rely on that dependency.

### PR DESCRIPTION
**Pull request by RomainMuller**
Fixes #2092

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -318,7 +318,7 @@ export class UserPool extends cdk.Construct implements IUserPool {
    */
   public readonly userPoolProviderUrl: string;
 
-  private triggers: CfnUserPool.LambdaConfigProperty = { };
+  private triggers: Mutable<CfnUserPool.LambdaConfigProperty> = { };
```
Comment (eladb) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts]: Interesting trick, but I actually found it refreshing to rewrite some of the code to rely on an the structure being immutable. The results were usually better and more readable code.

I am wondering also how this creeped up here.

File: packages/@aws-cdk/cdk/lib/stack.ts [hunk 2]
```diff
@@ -676,4 +676,4 @@ function findResources(roots: Iterable<IConstruct>): CfnResource[] {
 interface StackDependency {
   stack: Stack;
   reason: string;
-}
```
Comment (eladb) [inline packages/@aws-cdk/cdk/lib/stack.ts]: revert

File: packages/@aws-cdk/aws-serverless/package.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "@aws-cdk/aws-serverless",
+  "version": "0.26.0",
+  "deprecated": true,
```
Comment (eladb) [inline packages/@aws-cdk/aws-serverless/package.json]: is that an official npm thing?

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts (hunk 1 above)
Comment (RomainMuller) [reply packages/@aws-cdk/aws-cognito/lib/user-pool.ts]: Yeah this was actually not desired here... I'm looking to remove those spurious changes from the PR.

File: packages/@aws-cdk/aws-serverless/package.json (hunk 3 above)
Comment (RomainMuller) [reply packages/@aws-cdk/aws-serverless/package.json]: No it is not 😅

**Pull request merged**